app/services/submission/plant_trial_finalizer.rb

```diff
@@ -35,10 +35,80 @@ def create_new_trait_descriptors
 
   def create_scoring
     trait_mapping = submission.content.step04.trait_mapping
+    trait_scores = submission.content.step04.trait_scores || {}
+    accessions = submission.content.step04.accessions
+    replicate_numbers = submission.content.step04.replicate_numbers || {}
+    design_factors = submission.content.step04.design_factors || {}
+    design_factor_names = submission.content.step04.design_factor_names || []
+    lines_or_varieties = submission.content.step04.lines_or_varieties || {}
+
+    @new_plant_scoring_units = trait_scores.map do |plant_id, scores|
+      rollback(2) unless accessions[plant_id] && accessions[plant_id]['plant_accession'] && accessions[plant_id]['originating_organisation']
```

Comment:
I think that will be better because:

- we will be notified that sth went wrong by email
- user will not be left with submission failing for some hard to guess reason (probably a bug)

Optionally, if we want to be extra nice as this is an important step, we could provide custom error message for the action of finalizing submission, so that we ensure user that the data is safe and we will look into this issue.